src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -64,6 +64,20 @@
 			ReactorNetty.POOL_ACQUIRE_TIMEOUT,
 			"" + 45000));
 
+	/**
+	 * Default max idle time, fallback - max idle time is not specified.
+	 */
+	String DEFAULT_POOL_MAX_IDLE_TIME = System.getProperty(ReactorNetty.POOL_MAX_IDLE_TIME);
```

Comment:
I would maybe do the conversion to long here, with a default value of `"-1"` for the system property for instance, so that a misconfiguration is detected way earlier than the first time a resource is acquired from a pool.

Reply:
the conversion is done when constructing the Builder

> a quick side note on `maxIdleTime`: this is NOT really strictly enforced by the pool (eg. by a regular reaper job). Instead, the idle time of a resource is evaluated during `acquire()` and the resource is destroyed if it has been too idle (which at that point could be way over the maxIdleTime), and `acquire()` will look at the next resource.

that's OK

Reply:
The conversion happens when constructing the Builder https://github.com/reactor/reactor-netty/pull/1011/files#diff-838a952a6538b5173f40436cdfacae56R459

Reply:
yes, that's what I saw and what prompted me to make this comment. better to fail fast at JVM start than at a seemingly random point of app execution because the configuration error is detected late, the first time the `Builder` is constructed

Reply:
so I propose to detect the bad value for system property earlier than later, by doing the `Long.parseLong` here

Reply:
(if the property is not set, have the `String` be `"-1"` so that the `parseLong` doesn't fail but return `-1L`. down in `Builder` constructor, check `DEFAULT_POOL_MAX_IDLE_TIME != -1` to activate max idle time)

Reply:
I would say `DEFAULT_POOL_MAX_IDLE_TIME > -1`